monitorables/gitlab/api/usecase/gitlab.go

```diff
@@ -0,0 +1,289 @@
+	}
+}
+
+func (gu *gitlabUsecase) MergeRequestsGenerator(params interface{}) ([]uiConfigModels.GeneratedTile, error) {
+	prParams := params.(*models.MergeRequestGeneratorParams)
+
+	mergeRequests, err := gu.repository.GetMergeRequests(*prParams.ProjectID)
+	if err != nil {
+		return nil, &coreModels.MonitororError{Err: err, Message: "unable to load merge requests"}
+	}
+
+	var results []uiConfigModels.GeneratedTile
+	for _, mergeRequest := range mergeRequests {
+		p := &models.MergeRequestParams{}
+		p.ProjectID = prParams.ProjectID
+		p.ID = pointer.ToInt(mergeRequest.ID)
+
+		results = append(results, uiConfigModels.GeneratedTile{
+			Params: p,
+		})
+
+		// Add merge request into store
+		_ = gu.store.Set(gu.getMergeRequestStoreKey(*prParams.ProjectID, mergeRequest.ID), mergeRequest, mergeRequestCacheExpiration)
+	}
+
+	return results, nil
+}
+
+func (gu *gitlabUsecase) getProjectStoreKey(projectID int) string {
+	return fmt.Sprintf("%s:%s-%d", GitlabProjectStoreKeyPrefix, gu.repositoryUID, projectID)
+}
+
+// getProject load project information (from cache or api) and add result in cache
+func (gu *gitlabUsecase) getProject(projectID int) (*models.Project, error) {
+	project := &models.Project{}
+
+	storeKey := gu.getProjectStoreKey(projectID)
+	if err := gu.store.Get(storeKey, project); err != nil {
+		if project, err = gu.repository.GetProject(projectID); err != nil {
+			return nil, err
+		}
+
+		_ = gu.store.Set(storeKey, *project, projectCacheExpiration)
+	}
+
+	return project, nil
+}
+
+func (gu *gitlabUsecase) getMergeRequestStoreKey(projectID int, mergeRequestID int) string {
+	return fmt.Sprintf("%s:%s-%d-%d", GitlabMergeRequestStoreKeyPrefix, gu.repositoryUID, projectID, mergeRequestID)
+}
+
+// getMergeRequest load merge request information (from cache or api) and add result in cache
+func (gu *gitlabUsecase) getMergeRequest(projectID int, mergeRequestID int) (*models.MergeRequest, error) {
+	mergeRequest := &models.MergeRequest{}
+
+	storeKey := gu.getMergeRequestStoreKey(projectID, mergeRequestID)
+	if err := gu.store.Get(storeKey, mergeRequest); err != nil {
+		if mergeRequest, err = gu.repository.GetMergeRequest(projectID, mergeRequestID); err != nil {
+			return nil, err
+		}
+
+		_ = gu.store.Set(storeKey, *mergeRequest, mergeRequestCacheExpiration)
+	}
+
+	return mergeRequest, nil
+}
+
+func parseStatus(status string) coreModels.TileStatus {
+	// See: https://docs.gitlab.com/ee/api/pipelines.html#list-project-pipelines
+	switch status {
+	case "running":
+		return coreModels.RunningStatus
+	case "pending":
+		return coreModels.QueuedStatus
+	case "success":
+		return coreModels.SuccessStatus
+	case "failed":
+		return coreModels.FailedStatus
+	case "canceled":
+		return coreModels.CanceledStatus
+	case "skipped":
+		return coreModels.CanceledStatus
+	case "created":
+		return coreModels.QueuedStatus
+	case "manual":
+		return coreModels.ActionRequiredStatus
+	default:
+		return coreModels.UnknownStatus
```

Comment:
Maybe group keys that return the same value?
Is there a way in golang to get multiple cases with the same return?

Reply:
I can, but i don't like it. It's not git diff friendly.

Reply:
look's like : 

```
func parseStatus(status string) coreModels.TileStatus {
	// See: https://docs.gitlab.com/ee/api/pipelines.html#list-project-pipelines
	switch status {
	case "running":
		return coreModels.RunningStatus
	case "pending", "created":
		return coreModels.QueuedStatus
	case "success":
		return coreModels.SuccessStatus
	case "failed":
		return coreModels.FailedStatus
	case "canceled", "skipped":
		return coreModels.CanceledStatus
	case "manual":
		return coreModels.ActionRequiredStatus
	default:
		return coreModels.UnknownStatus
	}
}```

Reply:
+1 for the git diff
what about just group them?

Reply:
I use the ordrer of this page : https://docs.gitlab.com/ee/api/pipelines.html#list-project-pipelines

Reply:
OK